subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -62,6 +63,8 @@ static void reg_start(struct bt_mesh_light_ctrl_srv *srv)
 	if (srv->reg && srv->reg->start) {
 		srv->reg->start(srv->reg);
 	}
+	atomic_clear_bit(&srv->flags, FLAG_AMBIENT_LUXLEVEL_SET);
+	atomic_set_bit(&srv->flags, FLAG_REG_RUNNING);
```

Comment:
`reg_start` is called from `ctrl_enable` and `light_ctrl_srv_start`. In both cases `srv->flags` should be reset. `FLAG_AMBIENT_LUXLEVEL_SET` can be set to true while controller is disabled if another report from a sensor comes during this time, but if so, why should we reset it here?

Not sure why we need a separate flag. Regulator always works (assuming that a sensor sends reports) if the controller is enabled. Transitions can't happen if the controller is disabled. So when we call `transition_start`, the controller is enabled and `FLAG_REG_RUNNING` will always be (or should be by design) true.

Reply:
Controller FSM stops when manual level/ll messages are received. Then it can be turned on again by explicitly setting the mode=1.

Reply:
Regulator is also stopped in this case through `bt_mesh_light_ctrl_srv_disable` -> `ctrl_disable` -> `reg_stop` and flags are dropped in `ctrl_disable`.